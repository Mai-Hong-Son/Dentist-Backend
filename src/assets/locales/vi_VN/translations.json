{
  "modal": {
    "success_text": "OK"
  },
  "login": {
    "username": "Tên đăng nhập",
    "password": "Mật khẩu",
    "error_title": "Lỗi đăngnhập",
    "forgot": "Quên mật khẩu",
    "submit": "Đăng nhập",
    "sologan": "Nha sĩ là phải giàu"
  },
  "forgot_password": {
    "email": "Nhập email để lấy lại mật khẩu",
    "remember": "Bạn nhớ mật khẩu ?",
    "login": "Đăng nhập",
    "resend_email": "Bạn chưa nhập được email ?",
    "resend": "Gửi lại",
    "submit": "Lấy mật khẩu",
    "screen_title": "Quên mật khẩu",
    "success_title": "Gửi mật khẩu thành công",
    "success_message": "Mật khẩu của bạn đã được gửi về email, vui lòng kiểm tra."
  },
  "systems": {
    "netinfo": {
      "offline": "Mất kết nối mạng",
      "online": "Mạng đã hoạt động trở lại"
    },
    "browser": {
      "toast_copy": "Toast copy",
      "copy": "Copy",
      "refresh": "Refresh",
      "share": "Share"
    },
    "request": {
      "api_method_not_found": "Api method_not_found",
      "timeout": "Timeout"
    }
  },
  "profile": {
    "screen_title": "Hồ sơ"
  },
  "library": {
    "screen_title": "Danh sách ảnh",
    "back": "Trở lại"
  },
  "sidebar": {
    "questions": "Câu hỏi tư vấn",
    "notifications": "Thông báo",
    "logout_link": "Đăng xuất"
  },
  "tabs": {
    "home": "Trang chủ",
    "question": "Câu hỏi tư vấn",
    "schedule": "Lịch hẹn",
    "service": "Dịch vụ",
    "my_account": "My account",
    "question_detail": "Consulting detail"
  },
  "home": {
    "title": "CỐ VẤN"
  },
  "questions": {
    "list_all": "Danh sách câu hỏi",
    "view_all": "Xem tất cả",
    "screen_title": "Danh sách câu hỏi",
    "status": {
      "all": "Tất cả",
      "answered": "Đã trả lời",
      "pending": "Chưa trả lời",
      "expired": "Quá hạn"
    },
    "filters": {
      "service": "Dịch vụ",
      "service_all": "Tất cả",
      "status": "Trạng thái",
      "status_all": "Tất cả",
      "calendar": "Loại lịch hẹn",
      "calendar_all": "Tất cả"
    },
    "done": "Done",
    "pending": "Pending"
  },
  "question_detail": {
    "screen_title": "Chi tiết tư vấn",
    "auto_consulting": "Nội dung tư vấn",
    "reason_button_title": "Kết luận",
    "content": "Nội dung tư vấn",
    "picture": "Ảnh chụp",
    "xray": "Ảnh X-Quang",
    "answer_of_dentist": "Câu trả lời của bác sĩ"
  },
  "conclusion": {
    "screen_title": "Kết luận tư vấn",
    "error": "Lỗi gửi tư vấn",
    "form": {
      "result": "Kết quả",
      "result_title": "Độ hô của bạn",
      "time_expected": "Thời gian dự kiến",
      "method": "Thủ thuật",
      "difficult": "Độ khó",
      "submit": "Gửi phản hồi",
      "preview": "Trước & sau",
      "protrusion": "Độ hô của răng"
    },
    "dialog_success": {
      "title": "Gửi phản hồi thành công",
      "content": "Chúng tôi sẽ gửi tư vấn của bạn đến khách hàng"
    }
  },
  "schedule_list": {
    "title": "Danh sách đặt lịch"
  },
  "notifications": {
    "screen_title": "Danh sách thông báo",
    "delete_all": "Xóa hết",
    "empty_text": "Bạn chưa có thông báo nào",
    "message1": "Khi bạn đặt câu hỏi tư vấn hoặc đặt lịch,",
    "message2": "kết quả của bạn sẽ được thông báo ở đây."
  }
}