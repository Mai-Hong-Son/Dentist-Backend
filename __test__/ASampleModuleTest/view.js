/**
 * Created by vjtc0n on 3/28/17.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloClient, ApolloProvider } from 'react-apollo';
import I18n from 'redux-i18n'
import I18nTheme from 'redux-theme-switchable'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import injectTapEventPlugin from 'react-tap-event-plugin';
import { AgGridReact } from 'ag-grid-react';

import { translations } from '../../../src/translations/translations'
import { themes } from '../../../src/themes/themes'

import ActiveCell from '../../../src/Modules/UserManagementPage/UI/React/components/ActiveCell'
import CreateUserComponent from '../../../src/Modules/UserManagementPage/UI/React/components/CreateUserComponent'
import {
  UserManagementPage
} from '../../../src/Modules/UserManagementPage/UI/React/containers/UserManagementPage'
import configureStore from '../../../src/store/configureStore';

const store = configureStore();
const client = new ApolloClient();
injectTapEventPlugin();


const data = {
  $id: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
  $key: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
  address: "QE",
  loginName: "khanhpg@gmail.com",
  phone: "123123",
  roleGroupId: "2",
  status: 1,
  userId: "20",
  userName: "KhanhABCDHGE",
  uuid: "da73dc90137f11e7913c13138a0e0b6f"
}

const rowData = [
  {
    $id: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
    $key: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
    address: "QE",
    loginName: "khanhpg@gmail.com",
    phone: "0",
    roleGroupId: "2",
    status: 1,
    userId: "20",
    userName: "KhanhABCDHGE",
    uuid: "da73dc90137f11e7913c13138a0e0b6f"
  }
]

const rowData2 = [
  {
    $id: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
    $key: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
    address: "QE",
    loginName: "khanhpg@gmail.com",
    phone: "0",
    roleGroupId: "2",
    status: 1,
    userId: "20",
    userName: "KhanhABCDHGE",
    uuid: "da73dc90137f11e7913c13138a0e0b6f"
  }
]

const UserManagement = new UserManagementPage({
  
})

const ActiveCellComponent = new ActiveCell({
  colDef: {
    cellRendererParams: {
      updateUser: (userInfo) => {UserManagement.state.rowData[0] = userInfo}
    }
  }
})

const CreateUser = new CreateUserComponent({
  createUser: (userInfo) => { UserManagement.state.rowData.push(userInfo) }
})

describe('Testing ActiveCell Component', () => {
  test('Test ActiveCell component', () => {
    const component = renderer.create(
      <MuiThemeProvider>
        <ApolloProvider store={store} client={client}>
          <I18nTheme themes={themes}>
            <I18n translations={translations}>
              <ActiveCell data={data}/>
            </I18n>
          </I18nTheme>
        </ApolloProvider>
      </MuiThemeProvider>
    )
    
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    
    //console.log(tree.children[2].props.onClick)
  
    tree.children[2].props.onClick()
    tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
  
  test('Test CreateUser component', () => {
    const component = renderer.create(
      <MuiThemeProvider>
        <ApolloProvider store={store} client={client}>
          <I18nTheme themes={themes}>
            <I18n translations={translations}>
              <CreateUserComponent />
            </I18n>
          </I18nTheme>
        </ApolloProvider>
      </MuiThemeProvider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('Testing onSubmitUser()', () => {
    UserManagement.state.rowData = rowData
    ActiveCellComponent.setState = () => null
    ActiveCellComponent.state.userInfo = data
    ActiveCellComponent.onSubmitUser()
    expect(UserManagement.state.rowData).toEqual([
      {
        $id: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
        $key: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
        address: "QE",
        loginName: "khanhpg@gmail.com",
        phone: "123123",
        roleGroupId: "2",
        status: 1,
        userId: "20",
        userName: "KhanhABCDHGE",
        uuid: "da73dc90137f11e7913c13138a0e0b6f"
      }
    ])
  })
  
  it('Testing onSubmitCreateUser()', () => {
    UserManagement.state.rowData = rowData2
    CreateUser.setState = () => null
    CreateUser.state.userInfo = data
    CreateUser.onSubmitCreateUser()
    expect(UserManagement.state.rowData).toEqual([
      {
        $id: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
        $key: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
        address: "QE",
        loginName: "khanhpg@gmail.com",
        phone: "0",
        roleGroupId: "2",
        status: 1,
        userId: "20",
        userName: "KhanhABCDHGE",
        uuid: "da73dc90137f11e7913c13138a0e0b6f"
      },
      {
        $id: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
        $key: "da73dc90137f11e7913c13138a0e0b6f|20|khanhpg@gmail.com|KhanhABCDHGE|1",
        address: "QE",
        loginName: "khanhpg@gmail.com",
        phone: "123123",
        roleGroupId: "2",
        status: 1,
        userId: "20",
        userName: "KhanhABCDHGE",
        uuid: "da73dc90137f11e7913c13138a0e0b6f"
      }
    ])
  })
  
  it('Testing onFormPress()', () => {
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.modalOpen = true}
    ActiveCellComponent.onFormPress()
    expect(ActiveCellComponent.state.modalOpen).toEqual(true)
  })
  
  it('Testing handleClose()', () => {
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.modalOpen = false}
    ActiveCellComponent.handleClose()
    expect(ActiveCellComponent.state.modalOpen).toEqual(false)
  })
  
  it('Testing handleChange()', () => {
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.userInfo.roleGroupId = "2"}
    ActiveCellComponent.handleChange({}, 1, 2)
    expect(ActiveCellComponent.state.userInfo.roleGroupId).toEqual("2")
  })
  
  it('Testing onTextChange(username)', () => {
    const event = {
      target: {
        id: "username"
      }
    }
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.userInfo.userId = "30"}
    ActiveCellComponent.onTextChange(event, "30")
    expect(ActiveCellComponent.state.userInfo.userId).toEqual("30")
  })
  
  it('Testing onTextChange(fullName)', () => {
    const event = {
      target: {
        id: "fullName"
      }
    }
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.userInfo.userName = "30"}
    ActiveCellComponent.onTextChange(event, "30")
    expect(ActiveCellComponent.state.userInfo.userName).toEqual("30")
  })
  
  it('Testing onTextChange(company)', () => {
    const event = {
      target: {
        id: "company"
      }
    }
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.userInfo.address = "30"}
    ActiveCellComponent.onTextChange(event, "30")
    expect(ActiveCellComponent.state.userInfo.address).toEqual("30")
  })
  
  it('Testing onTextChange(email)', () => {
    const event = {
      target: {
        id: "email"
      }
    }
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.userInfo.loginName = "30"}
    ActiveCellComponent.onTextChange(event, "30")
    // Should validate email here
    expect(ActiveCellComponent.state.userInfo.loginName).toEqual("30")
  })
  
  it('Testing onTextChange(phoneNumber)', () => {
    const event = {
      target: {
        id: "phoneNumber"
      }
    }
    ActiveCellComponent.setState = () => {ActiveCellComponent.state.userInfo.phone = "30"}
    ActiveCellComponent.onTextChange(event, "30")
    expect(ActiveCellComponent.state.userInfo.phone).toEqual("30")
  })
  
})

describe('Testing CreateUser Component', () => {
  it('Testing onFormPress()', () => {
    CreateUser.setState = () => {CreateUser.state.modalOpen = true}
    CreateUser.onFormPress()
    expect(CreateUser.state.modalOpen).toEqual(true)
  })
  
  it('Testing handleClose()', () => {
    CreateUser.setState = () => {CreateUser.state.modalOpen = false}
    CreateUser.handleClose()
    expect(CreateUser.state.modalOpen).toEqual(false)
  })
  
  it('Testing handleChange()', () => {
    CreateUser.setState = () => {CreateUser.state.userInfo.roleGroupId = "2"}
    CreateUser.handleChange({}, 1, 2)
    expect(CreateUser.state.userInfo.roleGroupId).toEqual("2")
  })
  
  it('Testing onTextChange(username)', () => {
    const event = {
      target: {
        id: "username"
      }
    }
    CreateUser.setState = () => {CreateUser.state.userInfo.userId = "30"}
    CreateUser.onTextChange(event, "30")
    expect(CreateUser.state.userInfo.userId).toEqual("30")
  })
  
  it('Testing onTextChange(fullName)', () => {
    const event = {
      target: {
        id: "fullName"
      }
    }
    CreateUser.setState = () => {CreateUser.state.userInfo.userName = "30"}
    CreateUser.onTextChange(event, "30")
    expect(CreateUser.state.userInfo.userName).toEqual("30")
  })
  
  it('Testing onTextChange(company)', () => {
    const event = {
      target: {
        id: "company"
      }
    }
    CreateUser.setState = () => {CreateUser.state.userInfo.address = "30"}
    CreateUser.onTextChange(event, "30")
    expect(CreateUser.state.userInfo.address).toEqual("30")
  })
  
  it('Testing onTextChange(email)', () => {
    const event = {
      target: {
        id: "email"
      }
    }
    CreateUser.setState = () => {CreateUser.state.userInfo.loginName = "30"}
    CreateUser.onTextChange(event, "30")
    // Should validate email here
    expect(CreateUser.state.userInfo.loginName).toEqual("30")
  })
  
  it('Testing onTextChange(phoneNumber)', () => {
    const event = {
      target: {
        id: "phoneNumber"
      }
    }
    CreateUser.setState = () => {CreateUser.state.userInfo.phone = "30"}
    CreateUser.onTextChange(event, "30")
    expect(CreateUser.state.userInfo.phone).toEqual("30")
  })
})
