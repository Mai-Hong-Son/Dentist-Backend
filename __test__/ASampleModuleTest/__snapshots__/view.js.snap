// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ActiveCell Component Test ActiveCell component 1`] = `
<div
  className="row cell-container"
>
  <i
    className="col-lg-3 material-icons active-icon"
  >
    fiber_manual_record
  </i>
  <div
    className="col-lg-6"
  />
  <div
    className="col-lg-3"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      keyboard_arrow_down
    </i>
  </div>
</div>
`;

exports[`Testing ActiveCell Component Test ActiveCell component 2`] = `
<div
  className="row cell-container"
>
  <i
    className="col-lg-3 material-icons active-icon"
  >
    fiber_manual_record
  </i>
  <div
    className="col-lg-6"
  />
  <div
    className="col-lg-3"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      keyboard_arrow_down
    </i>
  </div>
</div>
`;

exports[`Testing ActiveCell Component Test CreateUser component 1`] = `
<div>
  <div
    className="create-user-button"
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "transparent",
        "borderRadius": "50%",
        "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
               0 3px 10px rgba(0, 0, 0, 0.23)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <button
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "red",
          "border": 10,
          "borderRadius": "50%",
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 56,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": "bottom",
          "width": 56,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <i
            className="material-icons"
            style={
              Object {
                "color": "#ffffff",
                "fill": "#ffffff",
                "height": 56,
                "lineHeight": "56px",
              }
            }
          >
            add
          </i>
        </div>
      </div>
    </button>
  </div>
</div>
`;
